src/py/pyodide/__init__.py

```diff
@@ -64,20 +75,32 @@
     "create_once_callable",
     "create_proxy",
     "destroy_proxies",
-    "eval_code",
-    "eval_code_async",
-    "find_imports",
     "open_url",
     "register_js_module",
     "remove_event_listener",
-    "run_js",
     "set_interval",
     "set_timeout",
-    "should_quiet",
     "to_js",
     "unregister_js_module",
 ]
 
 
 def __dir__() -> list[str]:
     return __all__
+
+
+for name in DEPRECATED_LIST:
+    globals()[f"_deprecated_{name}"] = globals()[name]
+    del globals()[name]
+
+
+def __getattr__(name):
+    if name in DEPRECATED_LIST:
+        warn(
+            f"{name} has been moved to pyodide.{DEPRECATED_LIST[name]}. "
+            "Accessing it through the pyodide module is deprecated.",
+            DeprecationWarning,
+        )
+        globals()[name] = globals()[f"_deprecated_{name}"]
+        return globals()[name]
```

Comment:
Do you need to rename at all:  I would have thought that just returning `globals()[name]` with a warning would be enough?  Though yes, they seem to be doing it in https://peps.python.org/pep-0562/

Reply:
This prevents the warning from being raised more than once per input.